app/src/main/java/net/squanchy/search/view/ItemsAdapter.java

```diff
@@ -1,41 +1,135 @@
 package net.squanchy.search.view;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
 
-import net.squanchy.search.model.TitledList;
+import net.squanchy.search.SearchResults;
+import net.squanchy.search.view.SpeakerAdapter.ViewTypeId;
 import net.squanchy.speaker.domain.view.Speaker;
 
-import static net.squanchy.search.view.SpeakerAdapter.ViewTypeId;
-
 class ItemsAdapter {
 
-    private final List<TitledList<Speaker>> lists = new ArrayList<>();
+    // These values are "random", we count on them not clashing with the other
+    // random values that are used for non-hardcoded numeric IDs (for events
+    // and speakers). This is a reasonable assumption in the Long range.
+    private static final long ITEM_ID_EVENTS_HEADER = 100;
+    private static final long ITEM_ID_SPEAKERS_HEADER = 101;
 
-    int getTotalItemsCount() {
-        int size = 0;
-        for (TitledList list : lists) {
-            size += list.size();
-        }
-        return size + getAdditionalItemsCount();
+    private final SearchResults searchResults;
+
+    ItemsAdapter(SearchResults searchResults) {
+        this.searchResults = searchResults;
     }
 
-    private int getAdditionalItemsCount() {
-        return lists.size();
+    int totalItemsCount() {
+        if (searchResults.isEmpty()) {
+            return 0;
+        }
+
+        int totalEventItemsCount = totalCountForSectionIncludingHeaders(searchResults.events());
+        int totalSpeakerItemsCount = totalCountForSectionIncludingHeaders(searchResults.speakers());
+
+        return totalEventItemsCount + totalSpeakerItemsCount;
     }
 
     boolean isEmpty() {
-        return lists.isEmpty();
+        return searchResults.isEmpty();
     }
 
-    void addItems(TitledList<Speaker> list) {
-        lists.add(list);
-    }
+    @ViewTypeId
+    int viewTypeAtAbsolutePosition(int position) {
+        ensurePositionExists(position);
 
-    int getViewTypeAt(int position) {
-        if (position == 0) {
+        int totalEventItemsCount = totalCountForSectionIncludingHeaders(searchResults.events());
+
+        int adjustedPosition;
```

Comment:
can you extract this to a private method? something like `calculateAdjustedPositionFor(position, totalEventItemsCount)`